pkg/controller/workload/workload_event.go

```diff
@@ -62,6 +63,20 @@ type Endpoint struct {
 	portList    []*workloadapi.Port
 }
 
+type NetworkAddress struct {
+	Network string
+	Address uint32
+}
+
+type WorkloadStore struct {
+	ByUid  map[string]*workloadapi.Workload
+	ByAddr map[NetworkAddress]*workloadapi.Workload
```

Comment:
Do not expose maps directly, a GetByAddr() method can be provided and add a lock in that method